### ISSUE TITLE
Weighted average by sample count for evolution

### ISSUE THREAD

**Issue by martinthomson** [mozilla/telemetry-dashboard#219]
The graph is great, but the key piece of information is obscured by noise as there are more versions that contribute data.  See [this](http://mzl.la/1RMisc8) for example.  If all you want to see is the effect on the same value over time, averaging the values reported by Beta 44 with the values from Beta 43 is perfectly reasonable and should make the graph simpler and easier to understand.

**Comment by chutten** [comment]
There may be something we can do by sanitizing results when they come from less than a certain percentage of the population...

**Comment by Uberi** [comment]
We currently sanitize values that are less than 1% of the maximum submissions. Perhaps an adjustable sanitizing threshold would work?

**Comment by martinthomson** [comment]
I'm not seeing how changing the sanitization threshold would apply to this feature request.  The only effect I see is that it might allow you to use values that would otherwise be discarded (e.g., if the sample size from Beta 42 is low, you can still use it when combined with Beta 43 and 44).  Am I missing something?

(To be clear, I'm looking for something like this: https://ipv.sx/telemetry/general-v2.html?channels=beta%20release&measure=HTTP_RESPONSE_VERSION&target=20)

Note that this is only really useful when you graph based on submission date.  You can't often combine by build date, except in rare cases where release management have a bumper day.  I'd disable the option unless you are showing submission date.

**Comment by chutten** [comment]
Sorry. We skipped over the "average previous release's results" because, especially for release channel, results between versions aren't comparable. Instead we were aiming straight for "get rid of the noisy data". 

Removing data that comes on days that have less than X% of the peak sample count was just the obvious-to-us way of getting rid of that noise. "sanitize" is the name of the function that does this at present for X=1.

**Comment by martinthomson** [comment]
> We skipped over the "average previous release's results" because, especially for release channel, results between versions aren't comparable.

I don't think that this matters.  Yes, there are cases where this is true, but for a great number of the values that we measure on an ongoing basis there is no difference in how they are measured.  As an *option*, selected by a user who understands the limitations of their choice, this reduces the noise in the graph far more than mere sanitization would.

I have no opinion on the feature you are talking about.  That doesn't interest me particularly.
